psl/schema-ast/src/source_file.rs

```diff
@@ -1,11 +1,23 @@
 use std::sync::Arc;
 
+use serde::{Deserialize, Deserializer};
+
 /// A Prisma schema document.
 #[derive(Debug, Clone)]
 pub struct SourceFile {
     contents: Contents,
 }
 
+impl<'de> Deserialize<'de> for SourceFile {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: String = serde::de::Deserialize::deserialize(deserializer)?;
+        Ok(s.into())
+    }
+}
```

Comment:
This allows deriving `Multiple(Vec<(String, SourceFile)>)` from `Vec<(String, String)>` within the following:

```rust
#[derive(Deserialize, Debug)]
#[serde(untagged)]
pub(crate) enum SchemaFileInput {
    Single(String),
    Multiple(Vec<(String, SourceFile)>),
}
```

It also highlights that `SourceFile` is created from a `String`.